cli/src/pcluster/models/s3_bucket.py

```diff
@@ -148,12 +148,96 @@ def configure_s3_bucket(self):
             bucket_name=self.name,
             configuration={"Rules": [{"ApplyServerSideEncryptionByDefault": {"SSEAlgorithm": "AES256"}}]},
         )
-        deny_http_policy = (
-            '{{"Id":"DenyHTTP","Version":"2012-10-17","Statement":[{{"Sid":"AllowSSLRequestsOnly","Action":"s3:*",'
-            '"Effect":"Deny","Resource":["arn:{partition}:s3:::{bucket_name}","arn:{partition}:s3:::{bucket_name}/*"],'
-            '"Condition":{{"Bool":{{"aws:SecureTransport":"false"}}}},"Principal":"*"}}]}}'
-        ).format(bucket_name=self.name, partition=self.partition)
-        AWSApi.instance().s3.put_bucket_policy(bucket_name=self.name, policy=deny_http_policy)
+
+        bucket_policy = self._generate_bucket_policy()
+        AWSApi.instance().s3.put_bucket_policy(bucket_name=self.name, policy=json.dumps(bucket_policy))
+
+    def _generate_bucket_policy(self):
+        """Generate the complete bucket policy for the S3 bucket."""
+        partition = self.partition
+        bucket_arn = format_arn(partition, "s3", "", "", self.name)
+        bucket_arn_with_wildcard = f"{bucket_arn}/*"
+
+        # Generate DenyHTTP policy statement
+        deny_http_statement = {
+            "Sid": "AllowSSLRequestsOnly",
+            "Effect": "Deny",
+            "Principal": "*",
+            "Action": "s3:*",
+            "Resource": [bucket_arn, bucket_arn_with_wildcard],
+            "Condition": {"Bool": {"aws:SecureTransport": "false"}},
+        }
+
+        # Get the policy statements required for CloudWatch Logs export
+        logs_policy_statements = self._get_bucket_policy_for_cloudwatch_logs()
+
+        # Combine all policy statements
+        all_statements = [deny_http_statement] + logs_policy_statements
+
+        # Create the combined bucket policy
+        bucket_policy = {
+            "Version": "2012-10-17",
+            "Statement": all_statements,
+        }
+
+        return bucket_policy
+
+    def _get_bucket_policy_for_cloudwatch_logs(self):
```

Comment:
We should cover the generation of the policy with a unit test

Reply:
Done, but seems like it's unnecessary, due to it doesn't have logic that worth to test. The generated policy statements depend on the return value of `get_service_principal`, `format_arn`, etc.

Reply:
A unit test is necessary even if the function returns a static string because it protects from regressions.

Example: what if I change the bucket policy version? That string is static, but still I want my unit test to capture the mistake.

Makes sense?

Reply:
Makes sense